src/components/SpeechSynth/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState, JSX } from 'react';
+import { SpeechSynth } from 'springroll';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+import CodeBlock from '@theme/CodeBlock';
+
+interface CodeExampleParams {
+  newMessage?: string,
+  newPitch?: number,
+  newVolume?: number,
+  newVoice?: number,
+  newRate?: number,
+}
+
+/**
+ * Speech Synth Component - Simple demo page showing off SpringRoll's speech synth module. Referenced in docs/Examples/speechsynth.mdx
+ * @returns JSX.Element
+ */
+export default function SpeechSynthExample(): JSX.Element {
+
+  const [speaker] = useState(new SpeechSynth({}));
+  const [message, setMessage] = useState('');
+  const [volume, setVolume] = useState(speaker.volume);
+  const [pitch, setPitch] = useState(speaker.pitch);
+  const [rate, setRate] = useState(speaker.rate);
+  const [voice, setVoice] = useState(0);
+  const [codeExample, setCodeExample] = useState(`const speaker = new SpeechSynth({ \n  volume: ${volume
```

Comment:
You can replace this state variable and remove the `updateCodeExample` function if you have the string in a memo object:

```
const codeExample = useMemo(() => `const speaker = new SpeechSynth({
  volume: ${volume},
  rate: ${rate},
  pitch: ${pitch},
  voice: ${voice}
});

speaker.say('${message}');`, 
[volume, rate, pitch, voice, message]);`
```

That will give you an object that stays in sync with the state variables, it'll return a cached value until state changes, and it should cut down on a bit of repetition.

Reply:
Thank you! I knew there was a way to do that but just couldn't remember.